xivo_dao/resources/user/persistor.py

```diff
@@ -58,14 +59,33 @@ def count_all_by(self, column_name, criteria):
         query = query.group_by(column)
         return query.all()
 
-    def search(self, parameters):
+    def search(self, parameters, is_db_sort_limit=True):
         view = self.user_view.select(parameters.get('view'))
         query = view.query(self.session)
         if self.tenant_uuids is not None:
             query = query.filter(User.tenant_uuid.in_(self.tenant_uuids))
+
+        if not is_db_sort_limit:
+            order = parameters.pop('order', None)
+            limit = parameters.pop('limit', None)
+            offset = parameters.pop('offset', None)
+            reverse = False if parameters.pop('direction', 'asc') == 'asc' else True
+
         rows, total = self.user_search.search_from_query(query, parameters)
         users = view.convert_list(rows)
-        return SearchResult(total, users)
+        if is_db_sort_limit:
+            return SearchResult(total, users)
+        else:
+            result = sorted(
+                users,
+                key=lambda x: unicodedata.normalize('NFKD', x.__getattribute__(order)),
+                reverse=reverse,
+            )
```

Comment:
I am worried of the performance implications of this code. Do you have some numbers to compare the execution time between a python-collated and non-collated computation on a large number of users, say 5000?

On a side note, did you take a look at [PostgreSQL collation](https://www.postgresql.org/docs/13/collation.html)?

Reply:
@sduthil  I tested with 10K users: sorting in Python mainly takes +50% longer, compared to PostgreSQL sorting (0.32sec vs 0.21sec).

I have already skimmed through the documentation.

Reply:
OK, seems to me like an acceptable tradeoff of ease of implementation vs execution time. Thank you for the details :)